src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -1,211 +1,171 @@
-    componentWillUnmount() {
-        if (this.props.hideOnScroll) {
-            document.removeEventListener('scroll', this.onScroll, true);
-        }
-    }
-
-    render() {
-        const {
-            items,
-            switcher,
-            popupClassName,
-            switcherWrapperClassName,
-            popupPlacement,
-            size,
-            menuProps,
-            children,
-        } = this.props;
-
-        const switcherButton = switcher || this.renderDefaultSwitcher();
-
-        return (
-            <React.Fragment>
-                <div
-                    ref={this.anchorRef}
-                    className={b('switcher-wrapper', switcherWrapperClassName)}
-                    onClick={this.onSwitcherClick}
-                >
-                    {switcherButton}
-                </div>
-                <DropdownMenuPopup
-                    open={this.state.popupIsShowed}
-                    anchorRef={this.anchorRef}
-                    items={items}
-                    onMenuItemClick={this.onMenuItemClick}
-                    onClose={this.handleClose}
-                    popupClassName={popupClassName}
-                    placement={popupPlacement}
-                    size={size}
-                    menuProps={menuProps}
-                >
-                    {children}
-                </DropdownMenuPopup>
-            </React.Fragment>
-        );
-    }
-
-    private renderDefaultSwitcher() {
-        const {size, defaultSwitcherClassName, defaultSwitcherProps, icon} = this.props;
-
-        return (
-            <Button
-                view="flat"
-                size={size}
-                {...defaultSwitcherProps}
-                className={b('switcher-button', defaultSwitcherClassName)}
-                disabled={this.props.disabled}
-            >
-                {icon}
-            </Button>
-        );
-    }
-
-    private onMenuItemClick = (
-        event: React.MouseEvent<HTMLElement, MouseEvent>,
-        action: DropdownMenuItemAction<T>,
-    ) => {
-        action(event, this.props.data);
-        this.props.onMenuToggle();
-        this.hidePopup();
-    };
-
-    private onSwitcherClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
-        if (this.props.disabled) {
+    /**
+     * Custom content inside the menu popup.
+     */
+    children?: ReactNode;
+};
+
+export const DropdownMenu = <T,>({
+    items = [],
+    size = 'm',
+    icon = <Icon data={DotsIcon} />,
+    onMenuToggle,
+    hideOnScroll = true,
+    data,
+    disabled,
+    switcher,
+    switcherWrapperClassName,
+    defaultSwitcherProps,
+    defaultSwitcherClassName,
+    onSwitcherClick,
+    menuProps,
+    popupClassName,
+    popupPlacement,
+    children,
+}: DropdownMenuProps<T>) => {
+    const [isPopupShown, setPopupShown] = useState(false);
+    const anchorRef = useRef<HTMLDivElement | null>(null);
+
+    const handleSwitcherClick: MouseEventHandler<HTMLDivElement> = (event) => {
```

Comment:
Can we memoize this callback with `useCallback`?

Reply:
that would be an unnecessary overhead since this callback is used on a plain HTML element

Reply:
a callback unwrapped into `useCallback()` won't cause a re-render by itself

the callback will indeed be recreated each re-render, but memoizing it would be pointless since plain HTML elements won't check whether their event handler props have changed before re-rendering

see also this thread:
https://github.com/facebook/react/issues/17055